crypto/fipsmodule/ec/ec_nistp.h

```diff
@@ -45,26 +45,44 @@ typedef uint32_t ec_nistp_felem_limb;
 // providing an appropriate methods object.
 typedef struct {
   size_t felem_num_limbs;
-  void (*add)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a, const ec_nistp_felem_limb *b);
-  void (*sub)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a, const ec_nistp_felem_limb *b);
-  void (*mul)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a, const ec_nistp_felem_limb *b);
-  void (*sqr)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a);
-  ec_nistp_felem_limb (*nz)(const ec_nistp_felem_limb *a);
-} ec_nistp_felem_meth;
+  void (*felem_add)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a, const ec_nistp_felem_limb *b);
+  void (*felem_sub)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a, const ec_nistp_felem_limb *b);
+  void (*felem_mul)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a, const ec_nistp_felem_limb *b);
+  void (*felem_sqr)(ec_nistp_felem_limb *c, const ec_nistp_felem_limb *a);
+  ec_nistp_felem_limb (*felem_nz)(const ec_nistp_felem_limb *a);
 
-const ec_nistp_felem_meth *p256_felem_methods(void);
-const ec_nistp_felem_meth *p384_felem_methods(void);
-const ec_nistp_felem_meth *p521_felem_methods(void);
+  void (*point_dbl)(ec_nistp_felem_limb *x_out,
+                    ec_nistp_felem_limb *y_out,
+                    ec_nistp_felem_limb *z_out,
+                    const ec_nistp_felem_limb *x_in,
+                    const ec_nistp_felem_limb *y_in,
+                    const ec_nistp_felem_limb *z_in);
+  void (*point_add)(ec_nistp_felem_limb *x3,
+                    ec_nistp_felem_limb *y3,
+                    ec_nistp_felem_limb *z3,
+                    const ec_nistp_felem_limb *x1,
+                    const ec_nistp_felem_limb *y1,
+                    const ec_nistp_felem_limb *z1,
+                    const int mixed,
+                    const ec_nistp_felem_limb *x2,
+                    const ec_nistp_felem_limb *y2,
+                    const ec_nistp_felem_limb *z2);
 
-void ec_nistp_point_double(const ec_nistp_felem_meth *ctx,
+} ec_nistp_meth;
+
+const ec_nistp_meth *p256_felem_methods(void);
```

Comment:
If the methods contain more than `felem` now, maybe you'd want to rename them.

Reply:
right, thanks for noticing!